misc-utils/fincore.c

```diff
@@ -222,33 +224,48 @@ static int fincore_fd (struct fincore_control *ctl,
 	return rc;
 }
 
+static int64_t get_size(int fd, struct stat *sb) {
+        if (S_ISBLK(sb->st_mode)) {
+                // maybe it is a block device.
+                int64_t len;
+                if (0==ioctl(fd, BLKGETSIZE64, &len)) {
```

Comment:
There is already `blkdev_get_size()` from `include/blkdev.h`.
It supports more platforms and can also fall back to `BLKGETSIZE`.

Reply:
thank you for the feedback! I've updated to use blkdev_get_size().